# HG changeset patch
# User Rik <rik@octave.org>
# Date 1383455882 25200
#      Sat Nov 02 22:18:02 2013 -0700
# Node ID 2ff47214de72dc201b50670bcff186405e10e285
# Parent  cfe5297e0f3b1d2699277d9604898c67a6ddc99a
datestr.m: Accomodate Mac OS X odd support of %p in strftime.

* scripts/time/datestr.m: For %!tests that have an AM/PM indicator,
use toupper on string before comparing to expected value since Mac OS X
does not return uppercase am/pm.

diff --git a/scripts/time/datestr.m b/scripts/time/datestr.m
--- a/scripts/time/datestr.m
+++ b/scripts/time/datestr.m
@@ -304,19 +304,25 @@ endfunction
 %!assert (datestr (testtime,6), "12/18")
 %!assert (datestr (testtime,7), "18")
 %!assert (datestr (testtime,8), "Sun")
 %!assert (datestr (testtime,9), "S")
 %!assert (datestr (testtime,10), "2005")
 %!assert (datestr (testtime,11), "05")
 %!assert (datestr (testtime,12), "Dec05")
 %!assert (datestr (testtime,13), "02:33:17")
-%!assert (datestr (testtime,14), " 2:33:17 AM")
+## Mac OS X interprets %p parameter to strftime as lower case am/pm indicator.
+## Accomodate this, although no other UNIX-based OS does this.
+%!test
+%! obs = toupper (datestr (testtime,14));
+%! assert (obs, " 2:33:17 AM");
 %!assert (datestr (testtime,15), "02:33")
-%!assert (datestr (testtime,16), " 2:33 AM")
+%!test
+%! obs = toupper (datestr (testtime,16));
+%! assert (obs, " 2:33 AM");
 %!assert (datestr (testtime,17), "Q4-05")
 %!assert (datestr (testtime,18), "Q4")
 %!assert (datestr (testtime,19), "18/12")
 %!assert (datestr (testtime,20), "18/12/05")
 %!assert (datestr (testtime,21), "Dec.18,2005 02:33:17")
 %!assert (datestr (testtime,22), "Dec.18,2005")
 %!assert (datestr (testtime,23), "12/18/2005")
 %!assert (datestr (testtime,24), "18/12/2005")
